internal/services/apimanagement/api_management_gateway_host_name_configuration_resource.go

```diff
@@ -0,0 +1,190 @@
+package apimanagement
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/apimanagement/mgmt/2021-08-01/apimanagement"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/schemaz"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceApiManagementGatewayHostNameConfiguration() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceApiManagementGatewayHostNameConfigurationCreateUpdate,
+		Read:   resourceApiManagementGatewayHostNameConfigurationRead,
+		Update: resourceApiManagementGatewayHostNameConfigurationCreateUpdate,
+		Delete: resourceApiManagementGatewayHostNameConfigurationDelete,
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.GatewayHostNameConfigurationID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": schemaz.SchemaApiManagementChildName(),
+
+			"api_management_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ApiManagementID,
+			},
+
+			"gateway_name": schemaz.SchemaApiManagementChildName(),
+
+			"certificate_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validate.CertificateID,
+			},
+
+			"host_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"is_client_certificate_requested": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+			},
+
+			"is_http2_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Default:  true,
+				Optional: true,
+			},
+
+			"is_tls10_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+			},
+
+			"is_tls11_enabled": {
```

Comment:
is is redundant when the property ends in enabled, and likely requested to?

Reply:
```suggestion
			"request_client_certificate_enabled": {
				Type:     pluginsdk.TypeBool,
				Optional: true,
			},

			"http2_enabled": {
				Type:     pluginsdk.TypeBool,
				Default:  true,
				Optional: true,
			},

			"tls10_enabled": {
				Type:     pluginsdk.TypeBool,
				Optional: true,
			},

			"tls11_enabled": {
```

Reply:
@katbyte thanks for your feedback:). Fixed.